frontend/src/Frontend/Wallet.hs

```diff
@@ -190,20 +193,27 @@ makeWallet model conf = do
     onGenKey = _walletCfg_genKey conf
     onDelKey = _walletCfg_delKey conf
     onCreateWOAcc = _walletCfg_createWalletOnlyAccount conf
+    onImportEd25519 = _walletCfg_importEd25519Keypair conf
     refresh = _walletCfg_refreshBalances conf
 
   performEvent_ $ liftIO (putStrLn "Refresh wallet balances") <$ refresh
 
   rec
     onNewKey <- performEvent $ attachWith (createKey . nextKey) (current keys) onGenKey
     onWOAccountCreate <- performEvent $ attachWith (createWalletOnlyAccount . nextKey) (current keys) onCreateWOAcc
+    onNewImportEd25519 <- performEvent $ createEd25519Account <$> onImportEd25519
     newBalances <- getBalances model $ current keys <@ refresh
 
     keys <- foldDyn id initialKeys $ leftmost
       [ ffor onNewKey $ snocIntMap . SomeAccount_Account
       , ffor (_walletCfg_importAccount conf) $ snocIntMap . SomeAccount_Account
       , ffor onDelKey $ \i -> IntMap.insert i SomeAccount_Deleted
       , ffor onWOAccountCreate $ snocIntMap . SomeAccount_Account
+      -- TODO: This eats up indexes for non wallet keys, which could lead to some bad UX
+      -- making it even harder to recover a wallet. May be worth separating them in local
+      -- storage and zipping them together somehow. Does anything out of here rely on the
+      -- stability of the indexes?
+      , ffor onNewImportEd25519 $ snocIntMap . SomeAccount_Account
```

Comment:
- I agree that we should store these separately, especially if we're treating them as a legacy thing. 
- I don't know of anything that depends on the stability of the keys.